scripts/quests/sandoria/DRG_AF1_A_Craftmans_Work.lua

```diff
@@ -0,0 +1,133 @@
+-----------------------------------
+-- A Craftmans Work
+-- !addquest 0 94
+-----------------------------------
+-- Miaux !gotoid 17723442
+-- Easterrn Altepa ??? !gotoid 17244622
+-----------------------------------
+require('scripts/globals/items')
+require("scripts/globals/keyitems")
+require('scripts/globals/npc_util')
+require('scripts/globals/quests')
+require("scripts/globals/status")
+require('scripts/globals/interaction/quest')
+-----------------------------------
+local easternAltepaID = require("scripts/zones/Eastern_Altepa_Desert/IDs")
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.SANDORIA, xi.quest.id.sandoria.A_CRAFTSMAN_S_WORK)
+
+quest.reward =
+{
+    item     = xi.items.PEREGRINE,
+    fame     = 20,
+    fameArea = xi.quest.fame_area.SANDORIA,
+}
+
+quest.sections =
+{
+    -- Section: Quest available.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+                player:getMainJob() == xi.job.DRG and
+                player:getMainLvl() >= xi.settings.main.AF1_QUEST_LEVEL
+        end,
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Miaux'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:getVar("Quest[0][94]csSeen") == 0 then
+                        return quest:progressEvent(73)
+                    elseif player:getVar("Quest[0][94]csSeen") == 1 then --Declined First Offering.
+                        return quest:progressEvent(71)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [73] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                        quest:setVar(player, 'Prog', 1)
```

Comment:
Overall, this quest structure looks good.
However, I don't think you need to set the prog variable to 1 in here, since this is it's own section.
You could get away with just setting it to 1 when killing the nm

Reply:
the rationale for keeping the vars where they are at is because the previous writers had them setup that way. While I understand it may not be primary concern, those that have a player base and pull this will have a hell of a time comparing vars to where they are. With the above sql commands, those server owners can change the varnames and the progress is left the same.